ggshield/core/errors.py

```diff
@@ -66,6 +66,15 @@ def __init__(self, message: str):
         super().__init__(ExitCode.UNEXPECTED_ERROR, message)
 
 
+class MissingScopeError(_ExitError):
+    """
+    Token does not have the required scope
+    """
+
+    def __init__(self, message: str):
+        super().__init__(ExitCode.UNEXPECTED_ERROR, message)
```

Comment:
Maybe you can make this exception take the scope as argument and do the formatting itself? This would make it more reusable in other verticals like honeytoken (currently honeytoken commands do not check if the token has the right scope, they just fail and say "check your personal access token has the `honeytokens:write` scope.")

The wording would have to be a bit more generic though, something like "Your token needs the `{scope}` scope to perform this operation.".